### ISSUE TITLE
Incorrect # of Rows Displayed

### ISSUE THREAD

**Issue by garylfowler** [influxdata/ui#5901]
When querying large datasets using the new Script Editor, a situation occurred where the number of tables that were shown in the display was zero, even though there always has to be at least 1 and the actual data did have tables included.  See slack thread:  https://influxdata.slack.com/archives/C030G5783M3/p1664228908454359

The counts should match what is in the actual payload.

**Comment by wiedld** [comment]
A key example;
<img width="562" alt="Screen Shot 2022-09-26 at 3 33 42 PM" src="https://user-images.githubusercontent.com/10232835/192592505-5d6265f5-f0be-494e-af5c-8459707449cc.png">

And the downloaded csv showed 5 tables:
https://influxdata.slack.com/archives/C030G5783M3/p1664231673723699?thread_ts=1664228908.454359&cid=C030G5783M3

**Comment by wiedld** [comment]
Taken today, on new tools:
* same query, increasing time range
* expected: more time --> more data
* actual: more time --> less data
* Ends with `0 tables, 131985 rows`


https://user-images.githubusercontent.com/10232835/193160762-bbb30227-fd32-44c6-b57c-77043bc32a8b.mp4

**Comment by wiedld** [comment]
Next steps, going to compare:
* actual data returned from network call
* data downloaded via csv
* table and row counts in UI

**Comment by wiedld** [comment]
We have now the ability to separately (non-blocking) generate:
* the truncated, parsed data for the table. (with metadata)
* the metadata for the entire query size (table and row counts).

.

The problem is that currently, we only display 1 set of metadata. While trying to solve 2 problems: (1) showing the full query data shape, or (2) showing metadata for the results displayed. This leads us to always have a disconnect to one of these two: 
* option A:
   * match the metadata displayed to the truncated table data shown.
  * but then we get disconnect between the metadata counts, and the total data returned in a query. 
* option B:
  * match the metadata displayed, to the total amount of data returned in the query.
  * but then we get disconnect between the metadata counts, and the data shown in the table
     * (e.g. if the user paginates to the end, the tableID does not match the metadata) 

.

In summary, it feels like we are trying to solve 2 problems (e.g. total query metadata, versus description of shown data) in a single set of numbers. Which means will always have disconnect with something. How should we resolve? Should we pick only one approach, and more clearly communicate what the numbers means? Or should we be displaying both sets of metadata numbers?

**Comment by garylfowler** [comment]
My original reason for bringing up this issue is that, as a user, I was confused why I was seeing a fewer number of tables and rows when I was expanding my search to a larger time range.  The message that I have exceeded my limit and so the data is truncated flashed by so fast, I wasn't seeing it, and didn't necessarily know that my data was being truncated - and so this reduction in tables and rows was very confusing.   I think the most elegant solution in this case would be a combination of A and B and show something like:

Query Result:   400 tables, 260000 rows, 46 MB; max display result exceeded - truncated to 24 tables, 4000 rows, 28MB.

But..  we don't really have the screen real estate for this - so we might need do some Design work to make this possible.  Additionally, not sure if calculating all of this will slow down the operation at all.  If it does, it may not be worth it.  But if it doesn't - definitely something we could pursue in the future.

In the short term, I am fine with just showing:  Maximum Display Limit Exceeded, result truncated to 28MB (or whatever limit is set) - instead of showing the table and row counts in the instances where a limit has been exceeded.

**Comment by nathanielc** [comment]
>  In the short term, I am fine with just showing: Maximum Display Limit Exceeded, result truncated to 28MB (or whatever limit is set) - instead of showing the table and row counts in the instances where a limit has been exceeded.

I like this solution, it both avoids the confusion of the number changing and it avoids the commitment to wait for the query to finish in its entirety. If we show total numbers then we need to read all of the results. In many cases getting all of the result will be expensive and I am not convinced that showing untruncated totals is that important.

**Comment by wiedld** [comment]
Will put in place this^^ solution.


> it avoids the commitment to wait for the query to finish in its entirety

I would just note that the interim solution (code already there) yields twice, such that we are returning the parsed table data (with the truncated counts) without waiting for the final metadata total on the query.

**Comment by nathanielc** [comment]
> At the worst case tested (~290k rows) it was avg 2.6 seconds for the final metadata tally...without any expensive memory usage.

That make sense, however the query engine did have to do the work of the rest of the query. If the UI hangs up on the connection then the query engine can and does just stop doing the work and can move on to the next query (same goes for storage and other services down the stack). That is my main concern with waiting for the query to finish is we create a lot of load on the system in order to only gain the summary count data.

**Comment by garylfowler** [comment]
I actually didn’t personally see this one - Denise did.  So not sure I
would be able to reproduce but I can try.  If Denise thinks it isn’t an
issue any longer or is data related - I am fine with closing it.

**Comment by wiedld** [comment]
> I actually didn’t personally see this one - Denise did. So not sure I would be able to reproduce but I can try. If Denise thinks it isn’t an issue any longer or is data related - I am fine with closing it.

Was an issue. Now fixed (see the PR linked above)
